tiledb/api/src/query/read/output/strategy.rs

```diff
@@ -193,32 +193,53 @@ where
                 ),
                 Just(validity),
             )
-                .prop_map(|(data, mut offsets, mut validity)| {
-                    validity.iter_mut().for_each(|v| {
-                        v.iter_mut().take(offsets.len()).for_each(
-                            |v: &mut u8| {
+                .prop_map(
+                    move |(data, mut offsets, mut validity)| {
+                        /*
+                         * What does "empty" look like for arrow-shaped offsets?
+                         * Is it an empty offset buffer?
+                         * Or is it a single offset `[0]`?
+                         * The arrow `OffsetBuffer` doc suggests the latter, but is the tiledb core
+                         * implementation compliant with that?
+                         * A glance at the source code suggests it is not, i.e. if the number
+                         * of records is zero then the offsets will be empty.
+                         * So we must be able to generate empty offsets here too, and
+                         * upstream code must be ready.
+                         */
```

Comment:
The config name of "extra_offset" suggests to me that the offsets for var data should be the moral equivalent of `&[0u64]` and if its not that, then we should probably file a bug against core if that's what Arrow expects.

Reply:
I definitely agree, it does need some verification but I'll make sure to follow up on that.